### ISSUE TITLE
Background color of type tips

### ISSUE THREAD

**Issue by iBreadInside** [microsoft/vscode#141212]
Issue Type: <b>Bug</b>

Hello!
I have a problem with background color of type tips. Sometimes after VS Code is just opened back-color get a random value including white shades. How can I fix this?
![image](https://user-images.githubusercontent.com/66948071/150639774-648139ad-cc67-4aee-bea0-2db17f826f12.png)

VS Code version: Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:40:02.816Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 3600 6-Core Processor               (12 x 3600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.93GB (8.11GB free)|
|Process Argv|--crash-reporter-id f38645f2-eefb-46ad-a39d-68cd1ccfbecd|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (10)</summary>

Extension|Author (truncated)|Version
---|---|---
npm-intellisense|chr|1.4.0
path-intellisense|chr|2.7.0
vscode-eslint|dba|2.2.2
EditorConfig|Edi|0.16.4
rest-client|hum|0.24.6
vscode-styled-components|jpo|1.6.6
vscode-language-pack-ru|MS-|1.63.3
vscode-versionlens|pfl|1.0.9
node-pack|Swe|0.1.16
vscodeintellicode|Vis|1.2.15

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392cf:30425750
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30424896
pythonvs932:30410667
wslgetstarted:30426333
vscop804:30404766
vs360cf:30404996
vsrem710cf:30416617
vsbas813:30426126
vsic-wlc-tsjs:30426436

```

</details>

**Comment by jrieken** [comment]
Inside the `workbench.colorCustomizations`-section of your settings.json-file, do you have configured any styles like `editorInlayHint.XYZ`. Also, do you use a special theme?

**Comment by iBreadInside** [comment]
{
    "window.zoomLevel": 0,
    "javascript.updateImportsOnFileMove.enabled": "always",
    "editor.suggestSelection": "first",
    "vsintellicode.modify.editor.suggestSelection":
"automaticallyOverrodeDefaultValue",
    "typescript.updateImportsOnFileMove.enabled": "always",
    "typescript.inlayHints.parameterNames.enabled": "all",
    "typescript.inlayHints.variableTypes.enabled": true,
    "editor.bracketPairColorization.enabled": true,
    "typescript.inlayHints.enumMemberValues.enabled": true,
    "javascript.inlayHints.parameterNames.enabled": "all",
    "javascript.inlayHints.parameterNames.suppressWhenArgumentMatchesName":
false
}

That is my settings.json And my theme is Dark+ (default)

вт, 25 янв. 2022 г. в 10:49, Johannes Rieken ***@***.***>:

**Comment by jrieken** [comment]
Can you please use dev tools to check what the assigned styles are? The video below includes the steps

* F1 > Toggle Developer Tools
* select "Elements" and Element Selector button
* select an inlay hint and check its style

https://user-images.githubusercontent.com/1794099/151032750-cba93772-bd1a-4554-bd41-20d72b7f3d35.mov

**Comment by iBreadInside** [comment]
For now I have got yellow color with 
![image](https://user-images.githubusercontent.com/66948071/151201611-b71ec8ab-07c8-42b9-838d-197541447d49.png)

**Comment by jrieken** [comment]
@hediet This looks fishy, right? There is two rules named `dyn-rule-0` but the have different values

**Comment by jrieken** [comment]
@iBreadInside  We fixed an issue that reminds me of this. Can you try this with [VS Code Insiders](https://code.visualstudio.com/insiders/)?

**Comment by hediet** [comment]
That's almost certainly the non-global counter that you made global @jrieken ;)

**Comment by iBreadInside** [comment]
@jrieken Checking few components with Insidiers - no weird colors by now.
![image](https://user-images.githubusercontent.com/66948071/151212613-a521a30f-11fc-47ad-bc3d-0ca0345e2ce3.png)

 But I don't know the cause of it in VS Code. Btw, I found this in VS Code - 
![image](https://user-images.githubusercontent.com/66948071/151211719-18505c61-32f6-4e86-bc99-e028b3af0757.png)

Second .dyn-rule-0 have white color. Maybe somehow it became lower in cascade or got higher specification (hope I use correct terms).

**Comment by jrieken** [comment]
Each editor instance has a util that helps to create dynamic css rules, these rules use a numeric name scheme dyn-rule-0, dyn-rule-1, etc. Now, when you have more than one editor instance you have two instances of that utils and they creating colliding css rules (since all start at 0). I have pushed https://github.com/microsoft/vscode/commit/322176235ff1d018fb261ab81cbf18b2dc148d55 a few days back to prevent this from happening. Closing this issue as fixed
